### PR TITLE
Typst template smoke tests with data edge cases

### PR DESCRIPTION
**Pull request by marlonbaeten**
Edge case tests for our typst templates.

These tests are created to find errors in the typst templates by generating PDF files with various input values.

The input values are semi-randomly constructed with the intent to find errors in the handling of edge-cases (empty strings, None values, empty arrays, missing data, etc.)

Note that these tests do not validate that the correct data is rendered in the PDF.

Two errors were found and fixed:

- `backend/templates/model-na-14-2.typ` contained an undefined parameter (on the current `main`)
- Na 31-2 fails if the start date/time of the committee session is not set. I have added a check in the code that will verify the date/time is set before generating the PDF.

### CODE REVIEW

File: backend/src/report/api.rs [hunk 1]
```diff
@@ -136,7 +136,11 @@ async fn generate_and_save_files(
     let mut conn = pool.acquire().await?;
     let committee_session =
         crate::committee_session::repository::get(&mut conn, committee_session_id).await?;
-    if committee_session.status != CommitteeSessionStatus::DataEntryFinished {
+
+    // Only generate files if the committee session is finished and has a start date time
```
Comment (lama-d3v) [inline backend/src/report/api.rs:140]: does this check need to be added to the typst tests as well?

**Review by praseodym — APPROVED** [review]

File: backend/src/report/api.rs (hunk 1 above)
Comment (marlonbaeten) [reply backend/src/report/api.rs:140]: We tests the typst models with all possible values from the data structure: https://github.com/kiesraad/abacus/pull/2238/files#diff-cdf0cab3d9c666e0d5500c1463b1c6a0f12d6659215773b2557d7bb0967f84b8R216

These tests make sure our models will be robust in accepting different inputs, we do not want to do data validation within typst, that should be done in our Rust code.
